modules/io/src/video/vpVideoReader.cpp

```diff
@@ -277,37 +296,39 @@ This method enables to use the class as frame grabber.
 */
 void vpVideoReader::acquire(vpImage< vpRGBa > &I)
 {
-	if (!isOpen) {
-		open(I);
-	}
+  if (!isOpen)
+  {
+    open(I);
+  }
 
-	//getFrame(I,frameCount);
-	if (imSequence != NULL)
-	{
-		imSequence->acquire(I);
-    frameCount++; // next index
+  //getFrame(I,frameCount);
+  if (imSequence != NULL)
+  {
+    imSequence->acquire(I);
+    frameCount += frameStep; // next index // important change
   }
```

Comment:
`frameCount` is incremented in `vpDiskGrabber`. We should here use `vpDiskGrabber::getImageNumber()` rather than incrementing both sides.
A nice additional feature would be to allow the user to rewind the image sequence when we reach the last image, and go forward when we come back to the first image. That's why I propose rather the following code:

````
  if (imSequence != NULL)
  {
    imSequence->setStep(frameStep);
    imSequence->acquire(I);
    frameCount = imSequence->getImageNumber();
    if (frameCount > lastFrame) {
      frameCount = lastFrame;
      imSequence->setImageNumber(frameCount);
    }
    else if (frameCount < firstFrame) {
      frameCount = firstFrame;
      imSequence->setImageNumber(frameCount);
    }
  }
````